src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -2383,6 +2385,22 @@ public ContainerStats stats(final String containerId)
     }
   }
 
+  public void stats(final String containerId, final StatsHandler handler)
+      throws DockerException, InterruptedException {
+    final WebTarget resource = resource().path("containers").path(containerId).path("stats");
+
+    try {
+      requestAndTail(GET, handler, resource, resource.request(APPLICATION_JSON_TYPE));
+    } catch (DockerRequestException e) {
+      switch (e.status()) {
+        case 404:
```

Comment:
could you move this check into the `requestAndTail` method itself? That would then seem to remove the need to throw an exception only to catch it and re-throw it a single method up the stack

Reply:
It is doable, but then I would need to pass the `containerId` to `requestAndTail` in order to create the `ContainerNotFoundException`. Personally, I think it would smell a bit, as `requestAndTail` seems more "HTTP-focused", while `stats` seems "API-focused". That's not a strong opinion though. What do you think?